src/client/client.c

```diff
@@ -95,7 +95,13 @@ int main(int argc, char** argv)
     if(success)
     {
         const char * username = getenv("USER");
-        const char * hostname = getenv("HOSTNAME");
+        char hostname_tmp[256];
+        int hostname_status = gethostname(hostname_tmp, 256);
+        char* hostname = NULL;
```

Comment:
What is the purpose of the _tmp?  Can't you just use the array directly in the following code?

Reply:
There is some logic later on that determines if there is a valid hostname by checking if hostname is NULL. If I use the array directly I lose the ability to perform that check.

I suppose alternatively I could just perform the hostname check immediately after calling gethostname().